File: packages/digital-ink-recognition/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/digitalinkrecognition/classes/results/RecognizeResult.java
package io.capawesome.capacitorjs.plugins.mlkit.digitalinkrecognition.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.google.mlkit.vision.digitalink.common.RecognitionCandidate;
import com.google.mlkit.vision.digitalink.common.RecognitionResult;
import io.capawesome.capacitorjs.plugins.mlkit.digitalinkrecognition.interfaces.Result;

public class RecognizeResult implements Result {

    @NonNull
    private final RecognitionResult recognitionResult;

    public RecognizeResult(@NonNull RecognitionResult recognitionResult) {
        this.recognitionResult = recognitionResult;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSArray candidatesResult = new JSArray();
        for (RecognitionCandidate candidate : recognitionResult.getCandidates()) {
            JSObject candidateResult = new JSObject();
            candidateResult.put("text", candidate.getText());
            Float score = candidate.getScore();
            if (score != null) {
                candidateResult.put("score", score);
            }
            candidatesResult.put(candidateResult);
        }
        JSObject result = new JSObject();
        result.put("candidates", candidatesResult);
        return result;
    }
}

File: packages/digital-ink-recognition/ios/Plugin/Classes/Results/RecognizeResult.swift
import Foundation
import Capacitor
import MLKitDigitalInkRecognition

@objc public class RecognizeResult: NSObject, Result {
    let candidates: [DigitalInkRecognitionCandidate]

    init(candidates: [DigitalInkRecognitionCandidate]) {
        self.candidates = candidates
    }

    @objc public func toJSObject() -> AnyObject {
        var candidatesResult = JSArray()
        for candidate in candidates {
            var candidateResult = JSObject()
            candidateResult["text"] = candidate.text
            if let score = candidate.score {
                candidateResult["score"] = score.floatValue
            }
            candidatesResult.append(candidateResult)
        }
        var result = JSObject()
        result["candidates"] = candidatesResult
        return result as AnyObject
    }
}
